--- ntp-4.2.6p3/configure.ac.bero	2013-01-16 01:38:50.152079179 +0100
+++ ntp-4.2.6p3/configure.ac	2013-01-16 01:38:55.157019329 +0100
@@ -40,7 +40,7 @@ dnl the 'build' machine is where we run
 dnl the 'host' machine is where the resulting stuff runs.
 AC_DEFINE_UNQUOTED([STR_SYSTEM], ["$host"],
     [canonical system (cpu-vendor-os) of where we should run])
-AM_CONFIG_HEADER([config.h])
+AC_CONFIG_HEADERS([config.h])
 dnl AC_ARG_PROGRAM
 
 ntp_atom_ok=${ntp_atom_ok=no}
